tools/util_windows.go

```diff
@@ -0,0 +1,163 @@
+// +build windows
+
+package tools
+
+import (
+	"io"
+	"io/ioutil"
+	"os"
+	"unsafe"
+
+	"github.com/rubyist/tracerx"
+	"golang.org/x/sys/windows"
+)
+
+var (
+	availableClusterSize = []int64{64 * 1024, 4 * 1024} // ReFS only supports 64KiB and 4KiB cluster.
+	GiB                  = int64(1024 * 1024 * 1024)
+)
+
+// _FSCTL_DUPLICATE_EXTENTS_TO_FILE :
+// Instructs the file system to copy a range of file bytes on behalf of an application.
+//
+// https://docs.microsoft.com/windows/win32/api/winioctl/ni-winioctl-fsctl_duplicate_extents_to_file
+const _FSCTL_DUPLICATE_EXTENTS_TO_FILE = 623428
```

Comment:
Even if Windows names this with a leading underscore, we probably don't want to.